SphericalShell/OpenMC_Output_Processing/SS_Depletion_PostProcessing.py

```diff
@@ -0,0 +1,62 @@
+import openmc
+import openmc.deplete
+import numpy as np
+import matplotlib.pyplot as plt
+
+def extract_nuclides(dep_file_path, time_units, depletable_mat_index) : 
+    '''
+    Identifies nuclides from each depletion timestep and stores them in an array
+    
+    inputs: 
+        dep_file_path : path to .h5 file with OpenMC depletion simulation results
+        time_units : units ('s', 'd', 'min', 'h', 'a') used when retrieving depletion timesteps
+        depletable_mat_index : index of depletable material in OpenMC Materials object
+    '''    
+    dep_results = openmc.deplete.Results(filename=dep_file_path)
+    time_steps = dep_results.get_times(time_units = time_units)
+    nuclide_array = np.ndarray([], dtype = object)
+    stable_init_nuc = np.ndarray([], dtype = object)
+    
+    for step in range(len(time_steps)):
+        # Obtain a materials object with depletion information at each timestep
+        materials_object = dep_results.export_to_materials(step)[depletable_mat_index]
+        all_nuc = materials_object.get_nuclides()      
+        for nuclide in all_nuc :   
+            # Remove stable nuclides at beginning of operation
+            if step == 0:
+                half_life = openmc.data.half_life(nuclide)
+                if half_life == None:
+                    stable_init_nuc = np.append(stable_init_nuc, nuclide)
+            if nuclide not in nuclide_array and nuclide not in stable_init_nuc:        
+                nuclide_array = np.append(nuclide_array, nuclide) 
+                       
+    # Remove None element from array initialization    
+    nuclide_array = [nuc for nuc in nuclide_array if nuc is not None]
+    return nuclide_array, materials_object, dep_results, time_steps

```

Comment:
This only returns the `materials_object` from the last time step, is that intentional/necessary?

Reply:
From what I understand, it doesn't matter which time step `materials_object` is from, as it is only used in conjunction with `get_atoms()`, which gives nuclide concentration vs. time based on the material the specified nuclide came from. I see the same results if I specify a material id for `get_atoms` instead of a materials object.